SharedPackages/BrowserServicesKit/Sources/BrowserServicesKit/ContentScopeScript/UserContentController.swift

```diff
@@ -125,15 +125,22 @@ final public class UserContentController: WKUserContentController {
 
         // Install initial WKScriptMessageHandlers if any. Currently, no WKUserScript are provided at initialization.
         installUserScripts([], handlers: earlyAccessHandlers)
-
-        assetsPublisherCancellable = assetsPublisher.sink { [weak self, selfDescr=self.debugDescription] content in
-            Logger.contentBlocking.debug("\(selfDescr): 📚 received content blocking assets")
-            Task.detached { [weak self] in
-                let contentBlockingAssets = await ContentBlockingAssets(content: content)
-                await self?.installContentBlockingAssets(contentBlockingAssets)
+        // 1. receive UserContentControllerNewContent from assetsPublisher
+        // 2. prepare ContentBlockingAssets(content: userContentControllerNewContent) asynchronously
+        // 3. receive the ContentBlockingAssets from contentBlockingAssetsPublisher and install
+        let contentBlockingAssetsPublisher = PassthroughSubject<ContentBlockingAssets, Never>()
+        assetsPublisherCancellables = [
+            contentBlockingAssetsPublisher.receive(on: DispatchQueue.main).sink { [weak self] contentBlockingAssets in
+                self?.installContentBlockingAssets(contentBlockingAssets)
+            },
+            assetsPublisher.sink { [selfDescr=self.debugDescription] content in
+                Logger.contentBlocking.debug("\(selfDescr): 📚 received content blocking assets")
+                Task.detached {
+                    let contentBlockingAssets = await ContentBlockingAssets(content: content)
+                    contentBlockingAssetsPublisher.send(contentBlockingAssets)
+                }
             }
```

Comment:
the UserContentController was somehow not released during test because of the detached Task context - and asserting on deallocation check. After this refactoring the tests don‘t pass there anymore.